e2e/tests/caseManagementOrder_test.js

```diff
@@ -2,26 +2,108 @@ const config = require('../config.js');
+  assertDraftCMO(I, '1', '1 January 2020', withJudgeStatus);
+  assertDraftCMO(I, '2', '1 March 2020', withJudgeStatus);
+});
+
+Scenario('Judge makes changes to agreed CMO and seals', async (I, caseViewPage, uploadCaseManagementOrderEventPage, reviewAgreedCaseManagementOrderEventPage) => {
+  await I.navigateToCaseDetailsAs(config.judicaryUser, caseId);
+  await caseViewPage.goToNewActions(config.applicationActions.reviewAgreedCmo);
+  reviewAgreedCaseManagementOrderEventPage.selectCMOToReview('1 March 2020');
+  await I.retryUntilExists(() => I.click('Continue'), '#reviewCMODecision_decision');
+  I.see('mockFile.docx');
+  reviewAgreedCaseManagementOrderEventPage.selectMakeChangesToCmo();
+  reviewAgreedCaseManagementOrderEventPage.uploadAmendedCmo(config.testNonEmptyWordFile);
+  await I.completeEvent('Save and continue', {summary: 'Summary', description: 'Description'});
+  I.seeEventSubmissionConfirmation(config.applicationActions.reviewAgreedCmo);
+  caseViewPage.selectTab(caseViewPage.tabs.orders);
+  assertSealedCMO(I, '1', '1 March 2020');
+});
+
+Scenario('Judge sends agreed CMO back to the local authority', async (I, caseViewPage, reviewAgreedCaseManagementOrderEventPage) => {
+  await I.navigateToCaseDetailsAs(config.judicaryUser, caseId);
+  caseViewPage.selectTab(caseViewPage.tabs.draftOrders);
+  await I.startEventViaHyperlink(linkLabel);
+  I.see('mockFile.docx');
+  reviewAgreedCaseManagementOrderEventPage.selectReturnCmoForChanges();
+  reviewAgreedCaseManagementOrderEventPage.enterChangesRequested(changeRequestReason);
+  await I.completeEvent('Save and continue', {summary: 'Summary', description: 'Description'});
+  I.seeEventSubmissionConfirmation(config.applicationActions.reviewAgreedCmo);
+  caseViewPage.selectTab(caseViewPage.tabs.draftOrders);
+  assertDraftCMO(I, '1', '1 January 2020', returnedStatus);
+});
+
+Scenario('Local authority makes changes requested by the judge', async (I, caseViewPage, uploadCaseManagementOrderEventPage) => {
+  await I.navigateToCaseDetailsAs(config.swanseaLocalAuthorityUserOne, caseId);
+  caseViewPage.selectTab(caseViewPage.tabs.draftOrders);
+  assertDraftCMO(I, '1', '1 January 2020', returnedStatus);
+  await localAuthoritySendsAgreedCmo(I, caseViewPage, uploadCaseManagementOrderEventPage);
   I.seeEventSubmissionConfirmation(config.applicationActions.uploadCMO);
   caseViewPage.selectTab(caseViewPage.tabs.draftOrders);
-  I.seeInTab(['Draft Case Management Order 1', 'Order'], 'mockFile.docx');
-  I.seeInTab(['Draft Case Management Order 1', 'Hearing'], 'Case management hearing, 1 January 2020');
-  I.seeInTab(['Draft Case Management Order 1', 'Date sent'], dateFormat(new Date(), 'dd mmm yyyy'));
-  I.seeInTab(['Draft Case Management Order 1', 'Status'], 'With judge for approval');
-  I.seeInTab(['Draft Case Management Order 1', 'Judge'], 'Her Honour Judge Reed');
+  assertDraftCMO(I, '1', '1 January 2020', withJudgeStatus);
+  I.dontSeeElement(locate(`//p/a[text()="${linkLabel}"]`));
```

Comment:
it is to low level, xpath should not be in test, it is page object responsibility to know how to access html